src/main/java/org/elasticsearch/plugin/readonlyrest/IndexLevelActionFilter.java

```diff
@@ -143,4 +159,51 @@ private void sendNotAuthResponse(RestChannel channel) {
         channel.sendResponse(resp);
     }
 
+    class LoggerActionListener implements ActionListener<SearchResponse> {
+        private final String action;
+        private final ActionListener<SearchResponse> baseListener;
+        private final SearchRequest searchRequest;
+        private final RequestContext requestContext;
+        
+        LoggerActionListener(String action, ActionRequest searchRequest,
+                ActionListener<SearchResponse> baseListener,
+                RequestContext requestContext) {
+            this.action = action;
+            this.searchRequest = (SearchRequest)searchRequest;
+            this.baseListener = baseListener;
+            this.requestContext = requestContext;
+        }
+        
+        public void onResponse(SearchResponse searchResponse) {
+            logger.info(
+                    "search: {" +
+                    "ACT:" + action +
```

Comment:
Can we print here also the request id (from RequestContext)? So one can correlate the incoming request log and compare? this would be vital for tracing back problems with index rewriting.